flax/traverse_util.py

```diff
@@ -159,6 +164,45 @@ def unflatten_dict(xs, sep=None):
     cursor[path[-1]] = value
   return result
 
+def variables_path_partitions(
```

Comment:
This function doesn't do anything with variables or partitioning specifically?
This function maps over a flattened dict so I think it should have a name and docstring that explains that.